### PR TITLE
IFI-2013 PoC for remote web components (inspired by remote apps)

### PR DESCRIPTION
**Pull request by izaera**
This implements something similar to remote apps, but it renders web components instead of running random JavaScript and providing a callback API.

This PoC can be tested with any existing web component as long as it can be fetched using a public URL and the name of the custom tag is the same name you give to the remote web component entry in the CRUD. 

During the development of this PoC the [web component in the next comment](https://github.com/liferay-frontend/liferay-portal/pull/366#issuecomment-687090914) has been used by adding it to document library and getting its URL, but you can also place it in any other public web server (however, not in GitHub Gists as they are not served with the proper Content-Type and some browsers refuse to load it).

The attached web component renders a little menu with two items: one to navigate to the portal's home page, the other to open system settings.

### CODE REVIEW

**Comment by izaera** [comment]
```javascript
console.log('Enter TopMenu');

class TopMenu extends HTMLElement {
  connectedCallback() {
    if (this.shadowRoot) {
    	return;
    }
  
    const menuContainer = document.createElement('div');
    
    const menuItem1 = document.createElement('a');
    const menuItem2 = document.createElement('a');

    menuItem1.innerHTML = 'Home';
    menuItem1.href = "http://localhost:8080";
    
    menuItem2.innerHTML = 'System Settings';
    menuItem2.href = "http://localhost:8080/group/control_panel/manage?p_p_id=com_liferay_configuration_admin_web_portlet_SystemSettingsPortlet&p_p_lifecycle=0&p_p_state=maximized&p_v_l_s_g_id=37537"
    
    menuContainer.appendChild(menuItem1);
    menuContainer.appendChild(document.createElement("br"));
    menuContainer.appendChild(menuItem2);
    
    this.attachShadow({ mode: 'open' }).appendChild(menuContainer);
  }
}

window.customElements.define('top-menu', TopMenu);

console.log('Exit TopMenu');
```

File: modules/apps/remote-web-component/remote-web-component-admin-web/src/main/java/com/liferay/remote/web/component/admin/web/internal/portlet/RemoteWebComponentPortlet.java [hunk 1]
```diff
@@ -101,8 +103,16 @@ public void render(
 			PrintWriter printWriter = renderResponse.getWriter();
 
 			printWriter.print(
-				"<iframe src=\"" + _remoteWebComponentEntry.getUrl() +
-					"\"></iframe>");
+				"<script src=\"" + _remoteWebComponentEntry.getUrl() +
```
Comment (brunobasto) [inline modules/apps/remote-web-component/remote-web-component-admin-web/src/main/java/com/liferay/remote/web/component/admin/web/internal/portlet/RemoteWebComponentPortlet.java:106]: Interesting..

**Comment by wincent** [comment]
Just a thought about naming: when we sent the "remote-web-app" stuff to Brian he ended up wanting it to be "remote-app" instead because the repetition of "web" in things like "remote-web-app/remote-web-app-admin-web" etc was grating to him.

So even though these _are_ "web components" (just as remote apps are "web apps"), it is possible that Brian might want this to be called "remote-component" instead of "remote-web-component". I can see "remote-web-component-admin-web" grating on him in the same way.

I can't predict though; just wondering...

**Comment by jbalsas** [comment]
> So even though these are "web components" (just as remote apps are "web apps"), it is possible that Brian might want this to be called "remote-component" instead of "remote-web-component". I can see "remote-web-component-admin-web" grating on him in the same way.

Just for clarification... this is **not meant** to be forwarded, just a PoC for some Global Services work being done by @rotty3000

**Comment by jbalsas** [comment]
Closing here... this has gone through so many hoops by now that I don't think it makes sense to just keep this around 😉

**Comment by wincent** [comment]
I will be keeping https://github.com/wincent/liferay-portal/pull/366 open for sentimental reasons.

feelings

**Pull request closed without merging**